### PR TITLE
Add teleport logic to AssetsTransferApi

### PR DESCRIPTION
**Pull request by TarikGul**
Waiting on https://github.com/paritytech/asset-transfer-api/issues/136 to be resolved for `SystemToPara` but I will continue to work on the rest of the logic.

### CODE REVIEW

File: src/errors/checkXcmTxInputs.ts [hunk 1]
```diff
@@ -57,6 +57,7 @@ export const checkAssetIdInput = (
 			const isNativeChain =
 				chainInfo.specName.toLowerCase() === specName.toLowerCase();
 
+			// TODO: This logic breaks when we have 2 chains that share the same native tokens.
```
Comment (TarikGul) [inline src/errors/checkXcmTxInputs.ts:60]: @marshacb So this logic below has an edge case we didn't cover. Such as what if the 2 chains are statemine, and polkadot. It will fail since they both share native tokens. 

You can see the failing tests in the test suite for github actions.

Comment (TarikGul) [reply src/errors/checkXcmTxInputs.ts]: Hey so I removed it, as I think the check might be a little redundant. Let me know what you think. We can add it back in when we do specific checks on direction.

**Review by marshacb — APPROVED** [review]
lgtm!
